### PR TITLE
Declare autocorrect as unsafe for `RSpec/ReceiveMessages`

### PR DESCRIPTION
**Pull request by bquorning**
As suggested in #1677.

I am unsure if I should change `VersionChanged` for the cop too.

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [ ] Feature branch is up-to-date with `master` (if not - rebase it).
- [ ] Squashed related commits together.
- [ ] Added tests.
- [ ] Updated documentation.
- [ ] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [ ] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have modified an existing cop's configuration options:

- [ ] Set `VersionChanged: "<<next>>"` in `config/default.yml`.

### CODE REVIEW

**Review by john-h-k — APPROVED** [review]
LGTM

File: config/default.yml [hunk 1]
```diff
@@ -733,6 +733,7 @@ RSpec/ReceiveCounts:
 RSpec/ReceiveMessages:
   Description: Checks for multiple messages stubbed on the same object.
   Enabled: pending
+  SafeAutoCorrect: false
   VersionAdded: '2.23'
```
Comment (ydah) [inline config/default.yml:737]:
```suggestion
  VersionAdded: '2.23'
  VersionChanged: "<<next>>"
```

**Review by ydah — APPROVED** [review]
Thank you!

**Comment by ydah** [comment]
We can also follow up after this PR to make autocorrect as safe as possible 👍🏻

**Comment by pirj** [comment]
Historically I was always against marking cops as unsafe if they were buggy.
The only case when cops should be marked as unsafe is when it is unsafe by design.
Not temporarily. Not in a non-real far fetched scenario.

**Comment by ydah** [comment]
>Historically I was always against marking cops as unsafe if they were buggy.
The only case when cops should be marked as unsafe is when it is unsafe by design.
Not temporarily. Not in a non-real far fetched scenario.

I agree with you. However, I believe that the case mentioned in the next comment may be difficult to design.
https://github.com/rubocop/rubocop-rspec/issues/1677#issuecomment-1659274540
However, it may be "a non-realistic far fetched scenario."

**Comment by bquorning** [comment]
Ah sorry @pirj, I only just saw your comment now. I had just merged #1688 and was about to push this change as a bugfix release (but I’ll wait, pending this discussion).

> Not temporarily.

My idea was exactly to _temporarily_ mark the cop as unsafe, and then have a potential subsequent PR make the cop safe again. Can you elaborate on why you prefer not having temporarily unsafe cops?

**Comment by pirj** [comment]
There’s nothing more permanent than temporary 😅

**Comment by bquorning** [comment]
Ah yes, that is true. However in this case we have released a bug (claiming the autocorrect is safe) and the fastest way to fix it is to stop claiming the autocorrect is safe. Issue #1677 is still open, so we (someone) can pick it up and implement safe autocorrect.

Would it be ok if I proceed with the v2.23.1 release?

**Comment by pirj** [comment]
Please go ahead with the release.

**Pull request merged**